cdk/src/open-data-platform/lambda/write-demographic-data-handler.ts

```diff
@@ -1,158 +1,223 @@
+              `Opened connection. Active connections = ${++connectionsCount}`,
+          );
+        },
+        onConnectionClosed: () => {
+          console.log(
+              `Closed connection. Active connections = ${--connectionsCount}`,
+          );
+        },
+        onQueryStart: (_query, {text, values}) => {
+          console.log(
+              `${new Date().toISOString()} START QUERY ${text} - ${
+                  JSON.stringify(
+                      values,
+                      )}`,
+          );
+        },
+        onQueryResults: (_query, {text}, results) => {
+          console.log(
+              `${new Date().toISOString()} END QUERY   ${text} - ${
+                  results.length} results`,
+          );
+        },
+        onQueryError: (_query, {text}, err) => {
+          console.log(
+              `${new Date().toISOString()} ERROR QUERY ${text} - ${
+                  err.message}`,
+          );
+        },
       });
+      console.log('Finished connecting to database...');
+      resolve(pool);
+    } catch (error) {
+      reject(error);
     }
+  });
+}
 
-// Executes SQL statement argument against the MainCluster db.
-const executeSqlStatement = async (secretArn: string, resourceArn: string,
-                                   statement: string,
-                                   db: string = 'postgres'): Promise<Object> => {
-  return new Promise(async function (resolve, reject) {
-    const sqlParams = {
-      secretArn: secretArn,
-      resourceArn: resourceArn,
-      sql: statement,
-      database: db,
-      includeResultMetadata: true
-    };
+// Inserts all rows into the demographics table.
+async function insertRows(
+    db: Queryable, rows: DemographicsTableRow[]): Promise<any[]> {
+  const rowOptions:
+      BulkOperationOptions<'census_geo_id'|'total_population'|
+                           'black_percentage'|'white_percentage'> = {
+        database: db,
+        tableName: `demographics`,
+        columnTypes: {
+          census_geo_id: sql`VARCHAR`,
+          total_population: sql`REAL`,
+          black_percentage: sql`REAL`,
+          white_percentage: sql`REAL`,
+        },
+      };
+
+  return Promise.all([bulkInsert({
+    ...rowOptions,
+    columnsToInsert: [
+      `census_geo_id`, `total_population`, `black_percentage`,
+      `white_percentage`
+    ],
+    records: rows,
+  })]);
+}
+
+// Inserts all rows into the demographics table.
+async function deleteRows(db: Queryable): Promise<any[]> {
+  return db.query(sql`DELETE
+          FROM demographics
+          WHERE census_geo_id IS NOT NULL`);
+}
 
-    await rdsDataService.executeStatement(sqlParams,
-        function (err: Error, data: Object) {
-          if (err) {
-            console.log(err);
-            reject(err);
-          } else {
-            console.log('Data is: ' + JSON.stringify(data));
-            resolve(data);
-          }
+// Reads the S3 CSV file and returns [DemographicsTableRow]s.
```

Comment:
nit: let's use either // or /**/ for function comments (below you use the latter)